### PR TITLE
fix decouple metadata

### PR DESCRIPTION
**Pull request by dinhlongviolin1**
Resolve Avaiga/taipy-designer#129

### CODE REVIEW

File: frontend/taipy-gui/base/src/wsAdapter.ts [hunk 1]
```diff
@@ -43,9 +43,9 @@ export class TaipyWsAdapter extends WsAdapter {
                 taipyApp.clientId = id;
                 taipyApp.updateContext(taipyApp.path);
             } else if (message.type === "GMC") {
-                const mc = (message.payload as Record<string, unknown>).data as string;
-                window.localStorage.setItem("ModuleContext", mc);
-                taipyApp.context = mc;
+                const payload = message.payload as Record<string, unknown>;
+                taipyApp.context = payload.context as string;
+                taipyApp.metadata = JSON.parse((payload.metadata as string) || "{}");
```
Comment (FredLL-Avaiga) [inline frontend/taipy-gui/base/src/wsAdapter.ts]: is a surrounding try/catch worth it ?

**Review by FredLL-Avaiga — DISMISSED** [review]

File: taipy/gui/gui.py [hunk 2]
```diff
@@ -1116,10 +1116,15 @@ def __handle_ws_get_module_context(self, payload: t.Any):
...
+                # get metadata if there is one
+                metadata: t.Dict[str, t.Any] = {}
+                page = self._get_page(str(payload.get("path")))
+                if hasattr(page._renderer, "_metadata"):
+                    metadata = getattr(page._renderer, "_metadata", {})
                 self.__send_ws(
                     {
                         "type": _WsType.GET_MODULE_CONTEXT.value,
-                        "payload": {"data": mc},
+                        "payload": {"context": mc, "metadata": json.dumps(metadata, cls=_TaipyJsonEncoder)},
```
Comment (FredLL-Avaiga) [inline taipy/gui/gui.py]: "metadata": json.dumps(metadata, cls=_TaipyJsonEncoder)} if metadata else None

File: frontend/taipy-gui/base/src/wsAdapter.ts (hunk 1 above)
Comment (dinhlongviolin1) [reply frontend/taipy-gui/base/src/wsAdapter.ts]: Yes you are right!

File: taipy/gui/gui.py (hunk 2 above)
Comment (dinhlongviolin1) [reply taipy/gui/gui.py]: metadata wouldnt be None since I set the default as an empty array

Comment (dinhlongviolin1) [reply taipy/gui/gui.py]: both the declaration and the default value of the getattr

Comment (FredLL-Avaiga) [reply taipy/gui/gui.py]: an empty dict would be falsy in python :-)

Comment (dinhlongviolin1) [reply taipy/gui/gui.py]: oh it is fine I want an empty dict anyway to ensure correct json parsing on the frontend side :D

**Comment by FredLL-Avaiga** [comment]
That was my point: to avoid parsing if not necessary ;-)

File: taipy/gui/gui.py (hunk 2 above)
Comment (dinhlongviolin1) [reply taipy/gui/gui.py]: Alright I will go with you :D

**Comment by FredLL-Avaiga** [comment]
Don't feel obliged
No big deal

**Review by FabienLelaquais — APPROVED** [review]
How come this works, I don't have a clue.
Well done, very certainly.

**Review by FredLL-Avaiga — APPROVED** [review]

**Pull request merged**